### PR TITLE
Fix inconsistent behaviour in REST Client reactive when using contexts

### LINKED ISSUE #32839: Inconsistent behaviour in REST Client Reative in regards of switching between Event Pool / Worker (inherited) executors
The REST Client Reactive is not using the Event Loop thread pool at the beginning, but the Worker thread pool (see [here](https://github.com/quarkusio/quarkus/blob/63cfe7c5589fc8d94e02df4fc25b3d3c42b10ca4/independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/AsyncInvokerImpl.java#L285)). 

Then, the handler ClientSendRequestHandler switches to the Event Loop thread [here](https://github.com/quarkusio/quarkus/blob/63cfe7c5589fc8d94e02df4fc25b3d3c42b10ca4/independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java#L106). 

And the problem is that if the response from the server arrives before the client ends with the ClientSendRequestHandler handler, the client will keep using the Worker thread pool. Why? 

1.- First, the ClientSendRequestHandler handler suspends the execution [here](https://github.com/quarkusio/quarkus/blob/63cfe7c5589fc8d94e02df4fc25b3d3c42b10ca4/independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java#L90).
2.- Second, the ClientSendRequestHandler handler performs the request to the server and subscribes to the response. After receiving the response, it resumes the execution [here](https://github.com/quarkusio/quarkus/blob/63cfe7c5589fc8d94e02df4fc25b3d3c42b10ca4/independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java#L347).

Now, we have two scenarios:

### Scenario A: Happy path
Because 1., the client suspends the execution and will wait for the resume. [This condition](https://github.com/quarkusio/quarkus/blob/63cfe7c5589fc8d94e02df4fc25b3d3c42b10ca4/independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/core/AbstractResteasyReactiveContext.java#L146) will be true and the while loop will end. 

After receiving the response from the server, the execution will resume using the Event Loop thread pool (provided by [this method](https://github.com/quarkusio/quarkus/blob/63cfe7c5589fc8d94e02df4fc25b3d3c42b10ca4/independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/core/AbstractResteasyReactiveContext.java#L122)).

### Scenario B: When server flies: The cause of the test failures

Before the client suspends the execution, if the server returns the response, the execution will be resumed because of 2. But because the execution was never suspended, it won't invoke the `getEventLoop()` method (see [here](https://github.com/quarkusio/quarkus/blob/63cfe7c5589fc8d94e02df4fc25b3d3c42b10ca4/independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/core/AbstractResteasyReactiveContext.java#L69)). Note that taking into account the execution was never suspended, I think this logic is correct. 

_Originally posted by @Sgitario in https://github.com/quarkusio/quarkus/issues/32644#issuecomment-1517643116_

### PR DESCRIPTION
**Pull request by Sgitario**
Note that I fixed the issue using this approach to avoid adding breaking changes.

Fix https://github.com/quarkusio/quarkus/issues/32839

### CODE REVIEW

**Review by geoand — APPROVED** [review]

**Comment by gsmet** [comment]
Should this be marked for backport?

**Comment by geoand** [comment]
No, as it is sort of a breaking change

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java [hunk 1]
```diff
@@ -88,30 +85,7 @@ public void handle(RestClientRequestContext requestContext) {
...
-                                public void handle(Void ignored) {
-                                    command.run();
-                                }
-                            });
-                        }
-                    }
-                });
+        Uni<HttpClientRequest> future = createRequest(requestContext);
 
         // DNS failures happen before we send the request
```
Comment (cescoffier) [inline independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java:90]: Do you know why we have this comment? Someone recently reported a DNS resolution failure when not on a duplicated context.

**Review by cescoffier — APPROVED** [review]

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java (hunk 1 above)
Comment (Sgitario) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java:90]: It was added as part of the MicroProfile 4 and Vert.x 4 upgrade a while ago: https://github.com/quarkusio/quarkus/pull/16333

Commit: https://github.com/quarkusio/quarkus/pull/16333/commits/0b88e3cc99e131a565de1a02aa8c819adce44fac

**Pull request merged**